common/djangoapps/student/models.py

```diff
@@ -1442,6 +1442,28 @@ def enrollments_for_user(cls, user):
         return cls.objects.filter(user=user, is_active=1)
 
     @classmethod
+    def preloaded_enrollments_for_user(cls, user):
+        """
+        List of user's CourseEnrollments, CourseOverviews preloaded if possible.
+
+        We try to preload all CourseOverviews, which are usually lazily loaded
+        as the .course_overview attribute. This is to avoid making an extra
```

Comment:
`._course_overview` is the real attribute that can store a `None` or the cached `CourseOverview`. But there's a getter property `.course_overview` that does the lazy loading as necessary.